lib/crypto/aead.lua

```diff
@@ -0,0 +1,94 @@
+--
+-- SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+--- AEAD (Authenticated Encryption with Associated Data) operations.
+-- This module provides a Lua wrapper for AEAD operations,
+-- using the underlying 'crypto_aead' C module.
+--
+-- @classmod crypto.aead
+
+local new = require("crypto_aead").new
+local sub = string.sub
+
+--- Prototype for AEAD instances.
+-- Objects of this type are created by `AeadModule.new()`.
+local AEAD = {}
+
+--- Closes the cipher instance and releases underlying C resources.
+-- If not called explicitly, this will be called by the garbage collector.
+function AEAD:close()
+	if self.tfm then self.tfm:__close() end
+	self.tfm = nil -- Allow the C object to be garbage collected by Lua
+end
+
+AEAD.__gc = AEAD.close
+AEAD.__close = AEAD.close
+AEAD.__index = AEAD
+
+--- Creates a new AEAD cipher instance.
+-- @function Aead.new
+-- @tparam string algname The algorithm name, e.g., "gcm(aes)".
+-- @treturn Aead An AEAD instance.
+-- @raise Error if C object creation fails.
+-- @usage
+--  local aead = require("crypto.aead")
+--  local gcm_aes = aead.new("gcm(aes)")
+function AEAD.new(algname)
+	return setmetatable({tfm = new(algname)}, AEAD)
+end
+
+--- Sets the encryption key.
+-- @tparam string key The encryption key.
+-- @raise Error if C operation fails.
```

Comment:
I think it's too vague.. not a blocker for merging, of course, but if we can be more specific here it would be better.. (quite minor comment)